ols/src/llms/providers/provider.py

```diff
@@ -43,7 +119,50 @@ def __init__(
         """
         self.model = model
         self.provider_config = provider_config
-        self.params = self._override_params(params or {})
+        params = self._override_params(params or {})
+        self.params = self._validate_parameters(params)
+
+    def _validate_parameters(self, params: dict[str, Any]) -> dict[str, Any]:
+        """Validate parameters for LLM provider."""
+        if self.provider_config is None:
+            logger.warning("Provider is not set. Parameters validation is disabled.")
+            return params
+
+        provider = self.provider_config.type
+
+        if provider is None:
+            logger.warning(
+                "Provider type is not set. Parameters validation is disabled."
+            )
+            return params
+
+        if provider not in available_provider_parameters:
+            logger.warning(
+                f"Available parameters for provider {provider} are not defined."
+                "Parameters validation is disabled."
+            )
+            return params
+
+        # retrieve all available parameters for provider
+        available_parameters = available_provider_parameters[provider]
+
+        # only supported parameters will be returned
+        filtered_params: dict[str, Any] = {}
+
+        for parameter_name, parameter_value in params.items():
+            parameter_type = type(parameter_value)
+            if parameter_value is None:
```

Comment:
This means that `"unknown_parameter": None` will go through the validation. Is this intended behavior?
Shoudn't it be inside the `if parameter not in available_parameters:` condition?

Reply:
ah actually it will be better, yes

Reply:
updated + added some unit tests